### ISSUE TITLE
l3draw: \draw_path_arc_axes:nnnn distorts following paths

### ISSUE THREAD

**Issue by jasperhabicht** [latex3/latex3#1195]
It seems that the `\draw_path_arc_axes:nnnn` macro of the `l3draw` package distorts the way the following paths are rendered.

Consider the following MWE (I took the values for the  `\draw_path_arc_axes:nnnn` macro from the example in the [package documentation](http://mirrors.ctan.org/macros/latex/contrib/l3experimental/l3draw.pdf) (on page 13) to make sure that I use it as intended):

```
\documentclass[border=10pt]{standalone}
\usepackage{l3draw}
\begin{document}
\ExplSyntaxOn
\draw_begin:

\draw_path_moveto:n { 0cm , 0cm }
\draw_path_lineto:n { 1cm , 1cm } 
\draw_path_arc_axes:nnnn { 0 } { 90 } { 2cm , 5mm } { 0cm , 1cm }  
\draw_path_use_clear:n { stroke }

\color_stroke:n { red }
\draw_path_moveto:n { 0cm , 0cm }
\draw_path_lineto:n { 1cm , 1cm } 
\draw_path_use_clear:n { stroke }

\draw_end:
\ExplSyntaxOff
\end{document}
```

This yields:

![image](https://user-images.githubusercontent.com/6378801/231879381-9e982ee7-3247-4387-81f8-0f438dff5a6f.png)

I would, however, expect the red line to exactly fit the black one drawn before, so that the result looks like in the following MWE which was constructed using PGF (thanks to @josephwright ):

```
\documentclass[border=10pt]{standalone}
\usepackage{tikz}
\begin{document}
\begin{tikzpicture}

  \pgfpathmoveto{\pgfpoint{0cm}{0cm}}
  \pgfpathlineto{\pgfpoint{1cm}{1cm}}
  \pgfpatharcaxes{0}{90}{\pgfpoint{2cm}{5mm}}{\pgfpoint{0cm}{1cm}}
  \pgfusepath{draw}
  
  \color{red}
  \pgfpathmoveto{\pgfpoint{0cm}{0cm}}
  \pgfpathlineto{\pgfpoint{1cm}{1cm}}
  \pgfusepath{draw}

\end{tikzpicture}
\end{document}
```

![image](https://user-images.githubusercontent.com/6378801/231879737-ed15f6db-c9e7-4b11-8c49-d9b69282a79b.png)

**Comment by muzimuzhi** [comment]
```tex
\documentclass{article}
\usepackage{l3draw}

\ExplSyntaxOn
\cs_gset_protected:Npn \draw_path_arc_axes:nnnn #1#2#3#4
  {
    \group_begin:
      \draw_transform_triangle:nnn { 0cm , 0cm } {#3} {#4}
      \draw_path_arc:nnn {#1} {#2} { 1pt }
    \group_end:
  }
\ExplSyntaxOff

\begin{document}
\ExplSyntaxOn
\draw_begin:

  \draw_path_moveto:n { 0cm , 0cm }
  \draw_path_lineto:n { 1cm , 1cm } 
  \draw_path_arc_axes:nnnn { 0 } { 90 } { 2cm , 5mm } { 0cm , 1cm } 
  \draw_path_use_clear:n { stroke }
  
  \color_stroke:n { red }
  \draw_path_moveto:n { 0cm , 0cm }
  \draw_path_lineto:n { 1cm , 1cm } 
  \draw_path_use_clear:n { stroke }

\draw_end:
\ExplSyntaxOff
\end{document}
```
![image](https://user-images.githubusercontent.com/6376638/231881820-a5cc58b1-e3c4-454a-9caf-24aa34b6b5c8.png)

<details><summary>Git diff</summary>
<p>

```diff
diff --git a/l3experimental/l3draw/l3draw-paths.dtx b/l3experimental/l3draw/l3draw-paths.dtx
index 70b7608..a4ec0b7 100644
--- a/l3experimental/l3draw/l3draw-paths.dtx
+++ b/l3experimental/l3draw/l3draw-paths.dtx
@@ -590,8 +590,10 @@
 %    \begin{macrocode}
 \cs_new_protected:Npn \draw_path_arc_axes:nnnn #1#2#3#4
   {
-    \draw_transform_triangle:nnn { 0cm , 0cm } {#3} {#4}
-    \draw_path_arc:nnn {#1} {#2} { 1pt }
+    \group_begin:
+      \draw_transform_triangle:nnn { 0cm , 0cm } {#3} {#4}
+      \draw_path_arc:nnn {#1} {#2} { 1pt }
+    \group_end:
   }
 %    \end{macrocode}
 % \end{macro}
```

</p>
</details>

**Comment by muzimuzhi** [comment]
Minor: It's `\draw_path_arc_axes:nnnn`, taking four `n`-args, but in issue title only three `n`-args were given.

**Comment by josephwright** [comment]
Fixed by 91504b0576597e74c6bdcd6bc3445dfa03f42fab

**Comment by josephwright** [comment]
@muzimuzhi Oops - sorted
